go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -397,6 +402,8 @@ func (na *NodeAllocator) Sync(nodes []interface{}) error {
 
 	networkName := na.netInfo.GetNetworkName()
 
+	nodeSubnets := na.clusterSubnetAllocator.ListAllIPv4Networks()
```

Comment:
Call this closer to where it is being used

Reply:
Also before iterating all the allocatable networks, it might be worth it to first check against `netInfo.Subnets()`

Reply:
We only need to call this function once. It returns all possible node subnets. Therefore, I don't think we can move it in the iteration below.